subsys/shell/shell_ops.c

```diff
@@ -412,8 +415,11 @@ void shell_vt100_color_set(const struct shell *shell,
 }
 
 void shell_vt100_colors_restore(const struct shell *shell,
-				       const struct shell_vt100_colors *color)
+				const struct shell_vt100_colors *color)
 {
-	shell_vt100_color_set(shell, color->col);
-	vt100_bgcolor_set(shell, color->bgcol);
+	if (IS_ENABLED(CONFIG_SHELL_VT100_COLORS)) {
```

Comment:
i think that it's not needed. Calls to this function (and `shell_vt100_colors_store`) in shell.c are surrounded by `if(CONFIG_SHELL_VT100_COLORS)` so if colors are off linker will remove those functions as unused.